v3/src/components/graph/components/graph-axis.tsx

```diff
@@ -44,16 +46,16 @@ export const GraphAxis = observer(function GraphAxis(
       return !!droppedAttrId
     }
   }
-
-  const {parentElt, wrapperElt,
-    setWrapperElt} = useAxisBoundsProvider(place, kGraphClassSelector)
-
   useDropHandler(droppableId, active => {
     const { dataSet, attributeId: droppedAttrId } = getDragAttributeInfo(active) || {}
     dataSet && droppedAttrId && isDropAllowed(place, dataSet, droppedAttrId) &&
       onDropAttribute?.(place, dataSet, droppedAttrId)
   })
 
+  useEffect(() => {
+    setParentElt(wrapperElt?.closest(kGraphClassSelector) as HTMLDivElement ?? null)
+  }, [wrapperElt])
+
```

Comment:
This isn't necessarily a problem, but note that this will require three renders before everything is set up:
- first render calls `setWrapperElt()`, which triggers the second render
- after the second render, this effect will fire, which calls `setParentElt()`, which triggers the third render
- third render will have everything in place

This could be reduced to two by determining the parent element during the first call to `setWrapperElt()`.

If you were interested in pursuing this, it would look something like this:
```typescript
  const parentEltRef = useRef<HTMLDivElement | null>(null)
  const [wrapperElt, _setWrapperElt] = useState<SVGGElement | null>(null)
  const setWrapperElt = useCallback((elt: SVGGElement | null) => {
    parentEltRef.current = elt?.closest(kGraphClassSelector) as HTMLDivElement ?? null
    _setWrapperElt(elt)
  }, [])
```